### PR TITLE
docs: initial content for UI Styling in NativeScript

### PR DESCRIPTION
**Pull request by Ombuweb**

### CODE REVIEW

File: content/guide/ui/styling.md [hunk 1]
```diff
@@ -0,0 +1,694 @@
+---
+title: Styling
+---
+
+In NativeScript you style the UI similarly to how you do it in a web application using Cascading Style Sheets (CSS), or by changing the style property of the views in JavaScript/Typescript. For CSS styling, NativeScript supports tools such as [Tailwind](https://docs.nativescript.org/plugins/tailwindcss.html), [SASS](#using-sass).
```
Comment (SeanKelly369) [inline content/guide/ui/styling.md]:
```suggestion
In NativeScript, UIs are styled similarly to how web applications using Cascading Style Sheets (CSS) are styled, or by changing the style property of the views in JavaScript/Typescript.  For CSS styling, NativeScript supports tools such as [Tailwind](https://docs.nativescript.org/plugins/tailwindcss.html), [SASS](#using-sass). 
```

Comment (SeanKelly369) [reply content/guide/ui/styling.md]: It is worth pointing out that SCSS and SASS can be used in place of CSS.

File: content/guide/ui/styling.md [hunk 2]
```diff
@@ -0,0 +1,694 @@
...
+
+In NativeScript the CSS styles can be set on 4 different levels:
+
+- [Application-wide CSS](#application-wide-css): Applies to every application page
+- [Page-specific CSS](#page-specific-css): Applies to the page's UI views
+<!-- - [Component-specific CSS](#component-specific-css): Applies for component only -->
+- [Inline CSS](#inline-css): Applies directly to a UI view
+- [Platform-specific CSS](#platform-specific-css)
+
+If there is CSS declared on different levels&mdash;all will be applied. The inline CSS will have the highest priority and the application CSS will have the lowest priority.
```
Comment (SeanKelly369) [inline content/guide/ui/styling.md]:
```suggestion
Like with web development, if there is CSS declared on different levels&mdash;all will be applied. The inline CSS will have the highest priority and the application CSS will have the lowest priority, with classes having a lower priority to IDs.  If multiple classes are called and they use the same properties, the last class called overwrites the previous property value.
```

File: content/guide/ui/styling.md [hunk 3]
```diff
@@ -0,0 +1,694 @@
...
+
+<!-- TODO: fix links -->
+
+| CSS Property          | JavaScript Property   | Description                                                                                                                                                                                                                               |
+| :-------------------- | :-------------------- | :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `color`               | `color`               | Sets a solid-color value to the matched view’s foreground.                                                                                                                                                                                |
+| `background`          | `background`          | Sets a solid-color value or [a linear gradient](https://docs.nativescript.org/cookbook/ui/styling) to the matched view’s background.                                                                                                      |
```
Comment (SeanKelly369) [inline content/guide/ui/styling.md]: Broken link, as mentioned on line 206

File: content/guide/ui/styling.md [hunk 4]
```diff
@@ -0,0 +1,694 @@
...
+| `letter-spacing`      | `letterSpacing`       | Sets the text letter spacing. (On Android API Level 21 and above.)                                                                                                                                                                        |
+| `line-height`         | `lineHeight`          | Sets the text line height                                                                                                                                                                                                                 |
+| `z-index`             | `zIndex`              | Sets the z-index. (On Android API Level 21 and above.)                                                                                                                                                                                    |
+| `clip-path`           | `clip-path`           | Sets the clip-path. Supported shapes are circle, ellipse, rect and polygon. You can define your own shape using [clippy](http://bennettfeely.com/clippy/)                                                                                 |
```
Comment (SeanKelly369) [inline content/guide/ui/styling.md]: Is it `clip-path`, or is it `clipPath`?

Comment (SeanKelly369) [reply content/guide/ui/styling.md]: There is no need for the / at the end of the link.

File: content/guide/ui/styling.md [hunk 5]
```diff
@@ -0,0 +1,694 @@
...
+| `android-dynamic-elevation-offset `    | `androidDynamicElevationOffset`    | **Android** | `View`        | Sets the elevation of the View in Android, which will be shown when an action was performed(e.g. `tap`, `touch`).                |
+| `off-background-color`                 | `offBackgroundColor`               | Both        | `Switch`      | Sets the background color of the Switch when it is turned off.                                                                   |
+| `highlight-color`                      | `highlightColor`                   | Both        | `TabStrip`    | Gets or sets the underline color of the selected `TabStripItem`.                                                                 |
+
+::: warning Note
+Currently, we can set only the `backgroundColor`, `color`, `fontFamily`, `fontSize`, `fontStyle`, `fontWeight` and `textTransform` styling properties to the `Label` and `Image` components inside the TabStripItem. More about the usage of those properties can be found in the [Supported CSS Properties](#supported-css-properties) section.
```
Comment (SeanKelly369) [inline content/guide/ui/styling.md:293]: Is this still the case?

File: content/guide/ui/styling.md [hunk 6]
```diff
@@ -0,0 +1,694 @@
...
+
+```xml
+<StackLayout class="home-panel">
+  <TextView
+    class="tvElevation"
+    editable="false"
+    textWrap="true"
+    text="TextView"
+  ></TextView>
+  <label androidElevation="5" class="sampleLabel" textWrap="true" text="Label"></label>
```
Comment (SeanKelly369) [inline content/guide/ui/styling.md]: Are the tags for `label` and `button` right?  Do they need to start with an uppercase character, or is it entirely optional?

File: content/guide/ui/styling.md [hunk 7]
```diff
@@ -0,0 +1,694 @@
...
+
+::: tip Note
+In native modals in Angular, the classes are applied to the first layout view in your modal component's HTML. If you are targeting a class that is applied to the root layout in your modal, you would target it with `.ns-dark.your-class`.
+:::
+
+## Supported Measurement Units
+
+NativeScript supports `DIPs` (Device Independent Pixels), `pixels` (via postfix px) and `percentages` (partial support for width, height and margin) as measurement units.
+
+NativeScript's recommended measurement unit is `DIP`. All measurable properties like width, height, margin, paddings, border-width, etc.) support `DIPs`. The font sizes are always measured in DIPs.
```
Comment (SeanKelly369) [inline content/guide/ui/styling.md:579]: It is worth mentioning DIPs is the default measurement in general.

File: content/guide/ui/styling.md (hunk 1 above)
Comment (ashutosh887) [inline content/guide/ui/styling.md]:
```suggestion
In NativeScript you style the UI similarly to how you do it in a web application using Cascading Style Sheets (CSS), or by changing the style property of the views in JavaScript/Typescript. For CSS styling, NativeScript supports tools such as [Tailwind](https://docs.nativescript.org/plugins/tailwindcss.html), [SASS](#using-sass). 
```

In NativeScript, UIs are styled similarly to how web applications using Cascading Style Sheets (CSS) are styled or by changing the style property of the views in JavaScript/TypeScript. For CSS styling, NativeScript supports tools such as [Tailwind](https://docs.nativescript.org/plugins/tailwindcss.html) and [SASS](#using-sass).

Comment (ashutosh887) [reply content/guide/ui/styling.md]: @NathanWalker

File: content/guide/ui/styling.md (hunk 2 above)
Comment (ashutosh887) [inline content/guide/ui/styling.md]:
```suggestion
If there is CSS declared on different levels&mdash;all will be applied. The inline CSS will have the highest priority and the application CSS will have the lowest priority.
```

If CSS is declared at different levels, all styles will be applied. Inline CSS takes the highest priority, while application-level CSS has the lowest priority.

**Review by ashutosh887 — CHANGES_REQUESTED** [review]
@NathanWalker I've added some comments to replace the current textual content there!

**Pull request merged**